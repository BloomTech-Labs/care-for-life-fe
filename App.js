import React from "react";
import Routes from "./src/routes";
import { Provider } from "react-redux";
import store from "./src/store";
import { persistStore } from "redux-persist";
import { PersistGate } from "redux-persist/es/integration/react";

const persistedStore = persistStore(store);

export default function App() {
  return (
    <Provider store={store}>
      <PersistGate persistor={persistedStore} loading={null}>
        <Routes />
      </PersistGate>
    </Provider>
  );
}
