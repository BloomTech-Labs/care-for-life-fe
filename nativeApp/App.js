import "react-native-gesture-handler";
import React, { useState, useEffect } from "react";
import { StyleSheet } from "react-native";
import { ApolloProvider } from "react-apollo";
import { ApolloOfflineProvider } from "react-offix-hooks";
import { offlineClient } from "./config/offix";
import SplashScreen from "./screens/SpashScreen";
import AppNavigator from "./navigation/AppNavigator";
import AuthNavigator from "./navigation/AuthNavigator";

const App = () => {
  const [initialized, setInitialized] = useState(false);
  const [token, setToken] = useState(null);

  // initialize the offix client and set the apollo client
  useEffect(() => {
    offlineClient.init().then(() => setInitialized(true));
  }, []);

  // load the app if the apolloClient is there, otherwise load the splash screen
  if (initialized) {
    return (
      <ApolloOfflineProvider client={offlineClient}>
        <ApolloProvider client={offlineClient}>
          {token === null ? (
            <AuthNavigator />
          ) : (
            <AppNavigator style={styles.container} />
          )}
        </ApolloProvider>
      </ApolloOfflineProvider>
    );
  }
  return <SplashScreen />;
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
});

export default App;
