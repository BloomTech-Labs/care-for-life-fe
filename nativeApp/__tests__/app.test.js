// __tests__/app.test.js
import React from 'react';
import App from '../App.js';
import { createHttpLink } from 'apollo-link-http';
import renderer from 'react-test-renderer';

test('Tests Running Correctly', async () => {
  expect(1).toBe(1);
});

test('App.js Renders Correctly', async () => {
  const tree = renderer.create(<App/>).toJSON();
  expect(tree).toMatchSnapshot();
});