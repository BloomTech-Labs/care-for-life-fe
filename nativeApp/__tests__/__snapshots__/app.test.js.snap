// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js Renders Correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 259.74,
        "width": 400,
      }
    }
  />
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;
